pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -270,13 +291,47 @@ func ingest(ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.
 	return errors.CombineErrors(ingestWithClient(), client.Close(ctx))
 }
 
+func ingestWithRetries(
+	ctx context.Context, execCtx sql.JobExecContext, ingestionJob *jobs.Job,
+) error {
+	ro := retry.Options{
+		InitialBackoff: 3 * time.Second,
+		Multiplier:     2,
+		MaxBackoff:     10 * time.Second,
+		MaxRetries:     5,
+	}
```

Comment:
One issue here is that a failure that happened 2 days ago feels like it should be irrelevant to a failure that happens today, but because nothing resets our retry counter, that failure matters forever in terms of both the backoff and the max count.

I wonder, is there a reason to ever stop retrying?  What's the advantage of going into a paused state vs retrying?  I suppose it means we stop forcing the source cluster to do work.